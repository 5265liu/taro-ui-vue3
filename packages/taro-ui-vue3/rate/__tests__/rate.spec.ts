import { mount } from '@vue/test-utils'
import AtRate from '../src/index.vue'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(AtRate, {
    slots,
    props: { ...values },
  })
}

describe('AtRate Snap', () => {
  it('render initial AtRate', () => {
    const wrapper = factory({ isTest: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtRate -- props className', () => {
    const wrapper = factory({ className: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtRate -- props customStyle', () => {
    const wrapper = factory({ customStyle: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtRate -- props size', () => {
    const wrapper = factory({ size: '10' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtRate -- props value', () => {
    const wrapper = factory({ value: 2 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtRate -- props max', () => {
    const wrapper = factory({ max: 10 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtRate -- props margin', () => {
    const wrapper = factory({ margin: 10 })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('AtRate Event', () => {
  it('AtRate onChange Event', () => {
    const onChange = jest.fn()
    const wrapper = factory({ value: 2, onChange: onChange })
    wrapper.find('.at-rate__icon').trigger('tap')
    expect(onChange).toBeCalled()
  })
})
