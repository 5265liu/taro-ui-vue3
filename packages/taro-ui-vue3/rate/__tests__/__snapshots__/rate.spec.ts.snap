// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtRate Snap render AtRate -- props className 1`] = `
<view
  class="at-rate test"
>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
</view>
`;

exports[`AtRate Snap render AtRate -- props customStyle 1`] = `
<view
  class="at-rate"
  style="color: red;"
>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
</view>
`;

exports[`AtRate Snap render AtRate -- props margin 1`] = `
<view
  class="at-rate"
>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
</view>
`;

exports[`AtRate Snap render AtRate -- props max 1`] = `
<view
  class="at-rate"
>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
</view>
`;

exports[`AtRate Snap render AtRate -- props size 1`] = `
<view
  class="at-rate"
>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
      style="font-size: 10px;"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
        style="font-size: 10px;"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
      style="font-size: 10px;"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
        style="font-size: 10px;"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
      style="font-size: 10px;"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
        style="font-size: 10px;"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
      style="font-size: 10px;"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
        style="font-size: 10px;"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
      style="font-size: 10px;"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
        style="font-size: 10px;"
      />
    </view>
  </view>
</view>
`;

exports[`AtRate Snap render AtRate -- props value 1`] = `
<view
  class="at-rate"
>
  <view
    class="at-rate__icon at-rate__icon--on"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--on"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
</view>
`;

exports[`AtRate Snap render initial AtRate 1`] = `
<view
  class="at-rate"
  isTest="true"
>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
  <view
    class="at-rate__icon at-rate__icon--off"
  >
    <view
      class="at-icon at-icon-star-2"
    />
    <view
      class="at-rate__left"
    >
      <view
        class="at-icon at-icon-star-2"
      />
    </view>
  </view>
</view>
`;
